geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManagerImpl.java

```diff
@@ -211,11 +211,17 @@ private PooledConnection createPooledConnection(ServerLocation serverLocation)
    */
   private PooledConnection forceCreateConnection(ServerLocation serverLocation)
       throws ServerRefusedConnectionException, ServerOperationException {
+    PooledConnection connection = null;
     connectionAccounting.create();
     try {
-      return createPooledConnection(serverLocation);
+      connection = createPooledConnection(serverLocation);
+      return connection;
```

Comment:
instead of the finally block, maybe add the `connectionAccounting.create()` line before the return?

Reply:
My original fix had the create in a finally if the connection was successfully created. During our meeting regarding this issue yesterday, it came up that this changed the behavior of the original code which was:

 1. increment the count before the connection is created
 2. create the connection

So, I changed the fix to do exactly what it used to do with the addition of the finally block to undo the increment if the connection failed.